### PR TITLE
build: add nix flake

### PR DESCRIPTION
**Pull request by isabelroses**
As per title.

Also should note this has backwards compatibility.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  pname = "kittysay";
+  version = "0.3.0";
+
+  src = ./.;
+  cargoLock.lockFile = ./Cargo.lock;
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "The cutest successor of cowsay";
```
Comment (uncenter) [inline default.nix:16]:
```suggestion
    description = "The cutest successor of cowsay.";
```

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 /target
+result
```
Comment (uncenter) [inline .gitignore:2]: Is this a directory?

Comment (isabelroses) [reply .gitignore:2]: directory, would you prefer I add a prepending slash.

Comment (isabelroses) [reply .gitignore]: completed in b017002bca8ad7eb3333a8ef8cf5a3949f3d1236

**Pull request merged**